File: api/Entities/Finance/VoucherItem.cs
using System.ComponentModel.DataAnnotations;

namespace api.Entities.Finance
{
    public class VoucherItem: BaseEntity
    {
        public VoucherItem()
        {
        }

        public int VoucherId { get; set; }
        public DateOnly TransDate { get; set; }
        public int COAId { get; set; }
        public string AccountName { get; set; }
        public long Dr {get; set;}
        public long Cr {get; set;}
        public string Narration { get; set; }
        public string DrEntryApprovedByAppUsername { get; set; }
        public DateOnly? DrEntryApprovedOn { get; set; }
        public bool DrEntryApproved { get; set; }
        public string Remarks { get; set; }
    
    }
}

File: client/src/app/_models/finance/voucherItem.ts


export interface IVoucherItem{
	id: number;
	financeVoucherId: number;
	transDate: Date;
	coaId: number;
	dr: number;
	cr: number;
	accountName: string;
	narration: string;
	drEntryApproved: boolean;
	drEntryApprovedByEmployeeId: number;
	drEntryApprovedOn: Date;
}

export class VoucherItem implements IVoucherItem
{
	id=0;
	financeVoucherId=0;
	transDate=new Date();
	coaId=0;
	accountName = '';
	dr=0;
	cr=0;
	narration= '';
	requiresApproval=false;
	drEntryApproved = false;
	drEntryApprovedByEmployeeId = 0;
	drEntryApprovedOn = new Date();
}